datatypes/src/raster/typed_raster.rs

```diff
@@ -91,96 +93,257 @@ where
+        None
+    }
+
+    pub fn get_f64_ref_mut(&mut self) -> Option<&mut BaseRaster<D, f64, Vec<f64>>> {
+        if let TypedRasterNDim::F64(r) = self {
+            return Some(r);
+        }
+        None
     }
 }
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, f32, Vec<f32>>
+// TODO: use a macro?
+// TODO: reactivate with trait specialization
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u8, Vec<u8>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::U8(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u16, Vec<u16>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::U16(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u32, Vec<u32>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::U32(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, u64, Vec<u64>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::U64(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i8, Vec<i8>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::I8(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i16, Vec<i16>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::I16(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i32, Vec<i32>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::I32(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, i64, Vec<i64>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::I64(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, f32, Vec<f32>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::F32(self)
+//     }
+// }
+//
+// impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, f64, Vec<f64>>
+// where
+//     D: GridDimension,
+// {
+//     fn into(self) -> TypedRasterNDim<D> {
+//         TypedRasterNDim::F64(self)
+//     }
+// }
+//
+impl<T> Into<TypedRasterNDim<Dim<[Ix; 2]>>> for BaseRaster<Dim<[Ix; 2]>, T, Vec<T>>
 where
-    D: GridDimension,
+    T: Pixel,
 {
-    fn into(self) -> TypedRasterNDim<D> {
-        TypedRasterNDim::F32(self)
+    fn into(self) -> TypedRasterNDim<Dim<[Ix; 2]>> {
+        T::get_typed_raster(self)
     }
 }
 
-impl<D> Into<TypedRasterNDim<D>> for BaseRaster<D, f64, Vec<f64>>
+impl<T> TryFrom<TypedRasterNDim<Dim<[Ix; 2]>>> for BaseRaster<Dim<[Ix; 2]>, T, Vec<T>>
```

Comment:
cant be ndim if you set it to "2"

Reply:
It's N dim with N=2, or do you mean I should use TypedRaster2D?

Reply:
I stuck with 2d for now 0c148c7dc15557c77a7dd6d7f9cd11c43545bfcd

Reply:
why can't you use a generic Dimension?

Reply:
I could but the rest of the code only uses 2d rasters

Reply:
well then it is ready for more dimensions in the future

Reply:
I generified it 1f4f5dd019bdf919624fbc059f71b396c701db6d